languages/python/exercises/concept/basics/.docs/after.md

```diff
@@ -0,0 +1,146 @@
+
+The `def` line is terminated with a colon. Statements for the _body_ of the function begin on the next line, and must be _indented in a block_. Functions explicitly return a value or object via the [`return`](https://docs.python.org/3/reference/simple_stmts.html#return) keyword. Functions that do not have an explicit `return` expression will return [`None`](https://docs.python.org/3/library/constants.html):
+
+```python
+#function definition on first line.
+def add_two_numbers(number_one, number_two):
+    return number_one + number_two  #returns the sum of the numbers
+
+>>> add_two_numbers(3, 4)
+7
+
+#this function has no return keyword, and so will return None
+def multiply_two_numbers(number_one, number_two):
+    result = number_one * number_two
+
+>>> print(multiply_two_numbers(6, 8))
+None
+```
+
+Functions are [called](https://docs.python.org/3/reference/expressions.html#calls) using their name followed by `()`. The number of arguments passed in the parentheses must match the number of parameters in the original function definition unless [default arguments](https://docs.python.org/3/tutorial/controlflow.html#default-argument-values) have been used:
+
+```python
+def number_to_the_power_of(number_one, number_two):
+    '''Returns float or int.
+
+       Takes number_one and raises it to the power of number_two, returning the result.
+    '''
+
+    return number_one ** number_two
+
+>>> number_to_the_power_of(3,3)
+27
+
+>>> number_to_the_power_of(4,)
+Traceback (most recent call last):
+  File "<stdin>", line 1, in <module>
+TypeError: number_to_the_power_of() missing 1 required positional argument: 'number_two'
+
+...
+
+def number_to_the_power_of_default(number_one, number_two=2):
+    '''Returns float or int.
+
+       Takes number_one and raises it to the power of number_two, returning the result.
+    '''
+
+    return number_one ** number_two
+
+>>> number_to_the_power_of_default(4)
+16
+```
+
+Methods bound to class names are invoked via _dot notation_ (`.`), as are functions, constants, or global names imported as part of a _module_.:
+
+```python
+import string
+
+#this is a constant provided by the *string* module
+>>> print(string.ascii_lowercase)
+"abcdefghijklmnopqrstuvwxyz"
+
+#this is a method call of the str *class*
+>>> start_text = "my silly sentence for examples."
+>>> str.upper(start_text)
+"MY SILLY SENTENCE FOR EXAMPLES."
+
+#this is a method call of an *instance* of the str *class*
+>>> start_text.upper()
+"MY SILLY SENTENCE FOR EXAMPLES."
+```
+
+[Comments](https://realpython.com/python-comments-guide/#python-commenting-basics) in Python start with a `#` (_hash character_) that is not part of a string, and end at line termination. Unlike many other programming languages, Python does not support multi-line comment marks. Each line of a comment block must start with the `#` character. Comments are ignored by the interpreter:
```

Comment:
I think leave the "hash character" bit out as people have different names for this.

```suggestion
[Comments](https://realpython.com/python-comments-guide/#python-commenting-basics) in Python start with a `#` that is not part of a string, and end at line termination. Unlike many other programming languages, Python does not support multi-line comment marks. Each line of a comment block must start with the `#` character. Comments are ignored by the interpreter:
```